### PR TITLE
Added descriptive minimum and maximum value labels to ORKScaleSliderView

### PR DESCRIPTION
**Pull request by edelabar**
This PR is to support a UI requirement that improves UX by directly associating numbers on the scale slider to their meaning rather than requiring the user to translate values in the Step's title or detail text.  

These changes were added to the ORKCatalog for demo purposes and screenshots are provided below.

Screenshots
-----------------

**Discrete Scale - Horizontal**
ios simulator screen shot may 8 2015 9 12 41 am

**Continuous Scale - Horizontal**
ios simulator screen shot may 8 2015 9 12 31 am

**Discrete Scale - Vertical**
ios simulator screen shot may 8 2015 9 12 21 am

**Continuous Scale - Vertical**
ios simulator screen shot may 8 2015 9 12 15 am

In addition to the visual changes, I also cleaned up the Auto Layout warnings in ec63648, and set the content mode of the ORKScaleSlider so it redraws correctly on orientation change in e1f5427.

### CODE REVIEW

**Comment by jwe-apple** [comment]
In addition to the above comments, we'll need JSON serialization support - look in ORKTest, run the tests, make sure they pass.

**Comment by jwe-apple** [comment]
Overall this is a great improvement! Thanks for doing this!

**Comment by jwe-apple** [comment]
I'll also be sending for review for RTL issues.

**Comment by edelabar** [comment]
@jwe-apple Thanks for the quick review, and for pointing out ORKTest, didn't realize it was there!  Should be all fixed up now.  I will definitely keep your comments in mind for future work.

**Comment by jwe-apple** [comment]
Awesome, thanks for the quick turnaround. I still don't see the JSON additions for the ORKTest tests (open ORKTest project, run the tests) -> look in ORKESerialization.[mh]. Also could you add some more cases to ORKTest like you had before for ORKCatalog, so we have both with and without labels?

**Comment by jwe-apple** [comment]
(I don't mind ORKTest having lots of cases in it, but ORKCatalog we're keeping clean)

**Comment by edelabar** [comment]
Got it. How's that?

**Comment by jwe-apple** [comment]
Looking good. I've added it to my testing queue.

ok to test

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 1]
```diff
@@ -270,6 +298,16 @@ ORK_CLASS_AVAILABLE
  */
 @property (readonly, getter=isVertical) BOOL vertical;
 
+/**
+ A localized label to describe the maximum value of the scale. (read-only)
+ */
+@property (readonly) NSString *maximumValueDescription;
```
Comment (jwe-apple) [inline ResearchKit/Common/ORKAnswerFormat.h]: Should be marked nullable, right?

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 2]
```diff
@@ -270,6 +298,16 @@ ORK_CLASS_AVAILABLE
...
+
+/**
+ A localized label to describe the minimum value of the scale. (read-only)
+ */
+@property (readonly) NSString *minimumValueDescription;
```
Comment (jwe-apple) [inline ResearchKit/Common/ORKAnswerFormat.h]: nullable, right?

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 3]
```diff
@@ -345,6 +404,16 @@ ORK_CLASS_AVAILABLE
...
```
Comment (jwe-apple) [inline ResearchKit/Common/ORKAnswerFormat.h]: These two should also be nullable

File: ResearchKit/Common/ORKAnswerFormat.h [hunk 4]
```diff
@@ -283,10 +321,31 @@ ORK_CLASS_AVAILABLE
...
  
  This method is the designated initializer.
  
+ @param maximumValue                The upper bound of the scale.
+ @param minimumValue                The lower bound of the scale.
+ @param defaultValue                The default value of the scale. If this value is out of range, the slider is displayed without a default value.
+ @param step                        The size of each discrete offset on the scale.
+ @param vertical                    Pass YES to use a vertical scale; for the default horizontal scale, pass NO.
+ @param maximumValueDescription     A localized label to describe the maximum value of the scale. For none, pass nil.
+ @param minimumValueDescription     A localized label to describe the minimum value of the scale. For none, pass nil.
```
Comment (jwe-apple) [inline ResearchKit/Common/ORKAnswerFormat.h]: `nil` and `NO` should be wrapped in back-ticks.

File: ResearchKit/Common/ORKScaleRangeDescriptionLabel.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKLabel.h"
+
+
+/**
+ *  Start/End description of the range in scale slider.
```
Comment (jwe-apple) [inline ResearchKit/Common/ORKScaleRangeDescriptionLabel.h]: No '*' at the beginning of the line, sorry for the nit!

**Comment by jwe-apple** [comment]
I felt the font of the labels was a little small when combined with a vertical slider. But happy to that accept is, and we'll review the UI while preparing for release. Just the few API-related nits above.

**Comment by jwe-apple** [comment]
And one last comment, from review of localizability:

UISliders don’t mirror their value when the userInterfaceLayoutDirection is right to left, but the Auto Layout constraints for the labels do. Instances of NSLayoutFormatDirectionLeadingToTrailing should change to NSLayoutFormatDirectionLeftToRight for horizontal sliders.

File: ResearchKit/Common/ORKScaleRangeDescriptionLabel.h (hunk 5 above)
Comment (edelabar) [inline ResearchKit/Common/ORKScaleRangeDescriptionLabel.h]: As in:
```objective-c
/**
 Start/End description of the range in scale slider.
 */
```
A lot of the other files in the Skin folder have the '*' before the body of the comment...

Comment (jwe-apple) [inline ResearchKit/Common/ORKScaleRangeDescriptionLabel.h]: True. We'll clean up at some point.

Comment (rsanchezsaez) [inline ResearchKit/Common/ORKScaleRangeDescriptionLabel.h]: *HeaderDoc* comments homogenized in [PR #181](https://github.com/ResearchKit/ResearchKit/pull/181).

**Comment by jwe-apple** [comment]
I'll give this one more test run tomorrow morning, but looks good to me.

**Comment by jwe-apple** [comment]
Tested. r=me.

**Pull request merged**